libs/smart-ngrx/src/selector/convert-children-to-virtual-array.function.ts

```diff
@@ -1,25 +1,30 @@
 import { EntityState } from '@ngrx/entity';
 
-import { ActionGroup } from '../actions/action-group.interface';
+import { actionFactory } from '../actions/action.factory';
 import { SmartNgRXRowBase } from '../types/smart-ngrx-row-base.interface';
 import { VirtualArrayContents } from '../types/virtual-array-contents.interface';
 import { VirtualArray } from './virtual-array.class';
+import { virtualArrayMap } from './virtual-array-map.const';
 
 /**
  * Converts the child field to a virtual array
  *
  * @param parentFieldName the name of the field in the row to convert
  * @param parentEntity the entity data we are dealing with
- * @param parentAction the action group for the row
+ * @param parentFeature the feature this entity belongs to
+ * @param parentEntityName the name of the entity
  */
 export function convertChildrenToVirtualArray<
   P extends SmartNgRXRowBase,
   C extends SmartNgRXRowBase,
 >(
   parentFieldName: keyof P,
   parentEntity: EntityState<P>,
-  parentAction: ActionGroup,
+  parentFeature: string,
+  parentEntityName: string,
 ): void {
+  const parentAction = actionFactory(parentFeature, parentEntityName);
+
   const length = parentEntity.ids.length;
   for (let i = 0; i < length; i++) {
```

Comment:
**suggestion (performance):** Evaluate performance impact of creating parentAction in function

While passing parentFeature and parentEntityName instead of parentAction provides more flexibility, it means that parentAction is now created inside the function. Consider measuring the performance impact of this change, especially if this function is called frequently. If it proves to be a bottleneck, you might want to consider memoizing the result of actionFactory.